### PR TITLE
Create the utils module and a test to verify all the functions

### PR DESCRIPTION
**Pull request by BradySkylight**
### **Description**

This PR is for the GCP team to create a Utils module to host all common functions that the various cloud functions may use.  This will provide a mechanism for cloud functions to be able to declare the module as a requirement and therefore have it installed for it's use at deploy.

Tests were created in the utils directory to ensure all functions are doing what is expected.

### CODE REVIEW

File: cloud-functions/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import logging
+import flask
+
+""" A Class to hold commonly used functions
+that can be reused by the different cloud functions
```
Comment (DanPaseltiner) [inline cloud-functions/utils/__init__.py:5]: nit: This is really just a collection of function in Python module, not a an actual class. I think this is totally fine, but we probably shouldn't call it a class.

File: cloud-functions/utils/__init__.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+TODO: We may want to move this into the SDK
```
Comment (DanPaseltiner) [inline cloud-functions/utils/__init__.py:7]: I think we can delete this.

**Review by DanPaseltiner — CHANGES_REQUESTED** [review]
This is really great. This will definitely make the development of additional Cloud Functions easier. There are a couple things I would like to see changed/discussed.

1. For the sake of consistency across all PHDI repos I think we should adopt the same documentation format and standards for these functions as in `phdi`. I'm not sure we need to go as far as generating docs with Sphinx, but I think user's will have an easier time reading our code if we are consistent.

2. I think we should consider returning a flask.Response object. I was noticing that the GCP logging still shows our functions finishing with 2XY status codes even when the code we have written logs and error or exception. GCP doesn't care that we are returning a JSON string with a key:value pair of `'status': 400`. However, if we return a flask.Response object and set the `.status = 400` I suspect GCP will reflect that in the logs. I would at least like to test this theory.

File: cloud-functions/utils/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Jinja2==3.1.2
+MarkupSafe==2.1.1
+msal==1.18.0
+msal-extensions==1.0.0
+msrest==0.7.1
+numpy==1.23.2
+oauthlib==3.2.0
+packaging==21.3
+pandas==1.4.3
+phdi @ git+https://github.com/CDCgov/phdi@ebdd43120eec138373bc36900cf6ea7e25aedc29
```
Comment (nickclyde) [inline cloud-functions/utils/requirements.txt:32]: Should this target `main` rather than a specific commit?

Comment (BradySkylight) [reply cloud-functions/utils/requirements.txt:32]: Eventually yes, but for now we are pointing to a specific version of the SDK project.  This isn't even necessary for the utils anyway.   I'll remove it for this requirements doc for now.

**Review by nickclyde — APPROVED** [review]
Looks great, thanks for moving this over Brady!

File: cloud-functions/utils/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+PyYAML==6.0
+requests==2.28.1
+requests-oauthlib==1.3.1
+rsa==4.9
+six==1.16.0
+smartystreets-python-sdk==4.11.6
+typing_extensions==4.3.0
+urllib3==1.26.11
+watchdog==2.1.9
+Werkzeug==2.2.2
```
Comment (DanPaseltiner) [inline cloud-functions/utils/requirements.txt:54]: Are all of these dependencies actually necessary for this module? This looks like everything that `phdi` depends on. Is this the result of `pip freeze >requirments.txt` in a virtual environment that has other stuff installed beyond what is needed here?

File: cloud-functions/utils/__init__.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    return result
+
+
+def _fail(message: str, status: str) -> flask.Response:
+    result = flask.Response(status=status, response=message)
+    result.status_code = 400
+    return result
+
+
+# Validate request header.
```
Comment (DanPaseltiner) [inline cloud-functions/utils/__init__.py:17]: With the update doc strings. I think we could probably loose this comment.

**Review by DanPaseltiner — APPROVED** [review]
Thanks for putting this together! LGTM

**Pull request merged**